plugins/woocommerce-admin/client/customize-store/assembler-hub/hooks/use-insert-pattern.ts

```diff
@@ -108,7 +108,20 @@ export const useInsertPattern = () => {
 		[ insertBlocks, insertableIndex, isActiveNewNeutralVariation ]
 	);
 
+	const removePattern = useCallback(
+		( pattern: BlockInstance ) => {
+			removeBlock( pattern.clientId, false );
+			trackEvent(
+				'customize_your_store_assembler_pattern_sidebar_click',
+				{ pattern: pattern.name }
+			);
```

Comment:
Should we identify this track event in some way to detect it was used to remove the pattern instead of inserting it?

Reply:
Yeah for sure, I think this was another thing that was raised as a requirement. Good catch @Aljullu!